lib/lightstep/tracer/span.rb

```diff
@@ -0,0 +1,132 @@
+# TODO(ngauthier@gmail.com) Separate Span and SpanContext under a getter according
+# to the spec. Baggage moves to span context.
+module LightStep
+  # Span represents an OpenTracer Span
+  #
+  # See http://www.opentracing.io for more information.
+  class Span
+    # UnsupportedValueTypeError is raised when a tag value is not a supported type
+    class UnsupportedValueTypeError < LightStep::Error; end
+
+    # The guid of the span
+    attr_reader :guid
```

Comment:
The Tracer needs to access it when starting a span:

https://github.com/lightstep/lightstep-tracer-ruby/blob/ngauthier/refactoring-and-spec-conformance/lib/lightstep/tracer/tracer.rb#L84

In ruby there's no package-privacy. So if Tracer and Span need to call methods on each other they must be public.

In ruby there's sort of a "make it public unless someone's going to break something" kind of policy.